src/SparqlEngineMain.cpp

```diff
@@ -60,11 +94,12 @@ int main(int argc, char** argv) {
   bool interactive = false;
   bool onDiskLiterals = false;
   bool allPermutations = false;
+  bool optimizeOptionals = true;
```

Comment:
Hannah said this currently slows down queries? I assume this only happens if there is an OPTIONAL? Is this the right default at the moment?

Reply:
The slowdown happens due to the query optimizer effectively having to optimize a query with another triple, representing the optional part. For queries with many triples (around 9 or more) the slowdown starts becoming noticeable. For queries with fewer triples the change in speed is minmal, and using the optimizer can of course lead to significantly faster execution time. If the default use case includes queries which, on average, have many triples changing the setting to false by default would be better.

Reply:
You are right, this case is actually more non-trivial than I thought.